persistence/trees/trees_test.go

```diff
@@ -1,19 +1,277 @@
+	"github.com/pokt-network/pocket/runtime/configs"
+	"github.com/pokt-network/pocket/runtime/test_artifacts"
+	"github.com/pokt-network/pocket/runtime/test_artifacts/keygen"
+	coreTypes "github.com/pokt-network/pocket/shared/core/types"
+	"github.com/pokt-network/pocket/shared/crypto"
+	"github.com/pokt-network/pocket/shared/messaging"
+	"github.com/pokt-network/pocket/shared/modules"
+	mockModules "github.com/pokt-network/pocket/shared/modules/mocks"
+	"github.com/pokt-network/pocket/shared/utils"
+	"github.com/stretchr/testify/assert"
+)
+
+var (
+	DefaultChains = []string{"0001"}
+
+	DefaultDeltaBig   = big.NewInt(100)
+	DefaultAccountBig = big.NewInt(1000000)
+	DefaultStakeBig   = big.NewInt(1000000000000000)
+
+	DefaultAccountAmount = utils.BigIntToString(DefaultAccountBig)
+	DefaultStake         = utils.BigIntToString(DefaultStakeBig)
+
+	DefaultStakeStatus     = int32(coreTypes.StakeStatus_Staked)
+	DefaultPauseHeight     = int64(-1) // pauseHeight=-1 implies not paused
+	DefaultUnstakingHeight = int64(-1) // unstakingHeight=-1 implies not unstaking
+
+	testSchema = "test_schema"
+
+	genesisStateNumValidators   = 5
+	genesisStateNumServicers    = 1
+	genesisStateNumApplications = 1
+)
 
-// TECHDEBT(#836): Tests added in https://github.com/pokt-network/pocket/pull/836
 func TestTreeStore_Update(t *testing.T) {
-	// TODO: Write test case for the Update method
-	t.Skip("TODO: Write test case for Update method")
+	pool, resource, dbUrl := test_artifacts.SetupPostgresDocker()
+	t.Cleanup(func() {
+		if err := pool.Purge(resource); err != nil {
+			log.Fatalf("could not purge resource: %s", err)
+		}
+	})
+
+	t.Run("should update actor tree and commit", func(t *testing.T) {
+		pmod := newTestPersistenceModule(t, dbUrl)
+		context := NewTestPostgresContext(t, 0, pmod)
+
+		actor, err := createAndInsertDefaultTestApp(context)
+		assert.NoError(t, err)
+
+		t.Logf("actor inserted %+v", actor)
+		hash, err := context.ComputeStateHash()
+		assert.NoError(t, err)
+		assert.NotEmpty(t, hash)
+	})
+
+	t.Run("compute state hash should fail if called after rollback", func(t *testing.T) {
+		pmod := newTestPersistenceModule(t, dbUrl)
+		context := NewTestPostgresContext(t, 0, pmod)
+
+		actor, err := createAndInsertDefaultTestApp(context)
+		assert.NoError(t, err)
+		assert.NotEmpty(t, actor)
+
+		hash1, err := context.ComputeStateHash()
+		assert.NoError(t, err)
+		assert.NotEmpty(t, hash1)
+
+		err = context.NewSavePoint()
+		assert.NoError(t, err)
+
+		actor2, err := createAndInsertDefaultTestApp(context)
+		assert.NoError(t, err)
+		assert.NotEmpty(t, actor2)
+
+		hash2, err := context.ComputeStateHash()
+		assert.NoError(t, err)
+		assert.NotEmpty(t, hash2)
+
+		context.RollbackToSavePoint()
+
+		_, err = context.ComputeStateHash()
+		assert.Error(t, err)
+	})
+}
+
+// createMockBus returns a mock bus with stubbed out functions for bus registration
+func createMockBus(t *testing.T, runtimeMgr modules.RuntimeMgr) *mockModules.MockBus {
```

Comment:
NIT: I would consider moving all these helpers to a file named `trees_utils_test.go`

Reply:
@dylanlott Thoughts on this?

Reply:
I'd rather create an issue with a `TECHDEBT` comment. There's no easy answer to the imports until we've solved some larger problems.